device/device.go

```diff
@@ -48,23 +49,20 @@ func Init(typeFlag string) {
 
 	fmt.Println("[+] Flashing", deviceType)
 
-	df, err := New(deviceType)
+	flasher, err := getFlasher(deviceType)
 	if err != nil {
 		fmt.Println("[-] Error: ", err)
 		return
 	}
-	if err := df.PrepareForFlashing(); err != nil {
-		fmt.Println("[-] Error: ", err)
-		return
-	}
-	if err := df.Configure(); err != nil {
+
+	if err := flasher.Flash(); err != nil {
 		fmt.Println("[-] Error: ", err)
 		return
 	}
 }
 
-// New triggers select repository methods and initializes a new flasher
-func New(device string) (Flasher, error) {
+// getFlasher triggers select repository methods and initializes a new flasher
+func getFlasher(device string) (Flasher, error) {
```

Comment:
__[gometalinter]__
* [gocyclo] cyclomatic complexity 18 of function getFlasher() is high (> 15)

<!-- Comment by [SideCI](https://sideci.com) -->